### ISSUE TITLE
add typedoc for better TS documentation

### ISSUE THREAD

**Issue by ctrlc03** [privacy-scaling-explorations/maci#754]
We can use TypeDoc to automatically generate documentation for all our TypeScript code in a nice and organized way, this is a HTML file which we can host on GitHub pages. This way, developers looking to use MACI can understand the codebase more easily, find the functions and classes they need, and see how to use them with minimal effort.

**Comment by baumstern** [comment]
How does adding typedoc enhance our TS documentation? Could you elaborate on its benefits?

**Comment by ctrlc03** [comment]
> How does adding typedoc enhance our TS documentation? Could you elaborate on its benefits?

have you looked at what TypeDoc is or tried cloning the PR code and running the command to generate the docs?
TLDR: We can use TypeDoc to automatically generate documentation for all our TypeScript code in a nice and organized way, this is a HTML file which we can host on GitHub pages. This way, developers looking to use MACI can understand the codebase more easily, find the functions and classes they need, and see how to use them with minimal effort.

**Comment by samajammin** [comment]
Exciting 😄 

@ctrlc03 what do you see as the scope of this issue - adding the TypeDoc code comments? Or generating the public page (e.g. deploying to GH pages via CI)? Or both?

**Comment by ctrlc03** [comment]
> Exciting 😄
> 
> @ctrlc03 what do you see as the scope of this issue - adding the TypeDoc code comments? Or generating the public page (e.g. deploying to GH pages via CI)? Or both?

I personally see this issue as more: add the package to the repo and the needed configuration files to make it work. 
Adding the actual comments on the functions and classes go into the refactoring of each package. 
Generating the public page is literally one command (npm run docs) -> for that imo we first need to complete this https://github.com/privacy-scaling-explorations/maci/issues/780 then it's just a matter of adding one instruction to the doc site deployment workflow.
